components/brave_shields/browser/cookie_pref_service.cc

```diff
@@ -21,6 +23,21 @@ namespace brave_shields {
 
 namespace {
 
+void SetCookieWhitelistForSocialMedia(HostContentSettingsMap* map,
+                                      PrefService* prefs) {
+  if (IntToContentSetting(prefs->GetInteger(
+          "profile.default_content_setting_values.cookies")) ==
+          ContentSetting::CONTENT_SETTING_ALLOW &&
+      prefs->GetBoolean(prefs::kBlockThirdPartyCookies)) {
+    // Only set social media exceptions when cross-site cookie blocking
+    // is enabled
+    ControlType type = prefs->GetBoolean(kGoogleLoginControlType)
+                           ? ControlType::ALLOW
+                           : ControlType::BLOCK;
+    SetCookieWhitelist(map, type, kGoogleOAuthPattern);
```

Comment:
I think it's problematic to set this as a brave cookie plugin type because it will always end up with an override one direction or the other which could potentially conflict with user overrides. It seems better to convert this directly to cookie settings in BravePrefProvider so we only add an override when it's enabled

Reply:
it's especially problematic because of this https://github.com/brave/brave-core/pull/3784/files#diff-6c160ccdf43d9f23e468a95ec08b4d12R219 which adds an allow override no matter what the setting is